### PR TITLE
UBTT_PlayAnimMontage bug fix

### PR DESCRIPTION
**Pull request by nozomanai**
- Excluding the need to create a new object every time a node is executed
- The montage works without ending after the node is interrupted
- Code optimization

I think this is the best version currently available

### CODE REVIEW

File: Source/UnrealHelperLibrary/Public/AI/Tasks/BTT_PlayAnimMontage.h [hunk 1]
```diff
@@ -7,37 +7,23 @@
...
-		if(Proxy->PlayMontage(InSkeletalMeshComponent, MontageToPlay, PlayRate, StartingPosition, StartingSection))
-		{
-			Proxy->SetFlags(RF_StrongRefOnFrame);
-			return Proxy;
-		}
-		
-		Proxy->MarkAsGarbage();
-		return nullptr;
-	}
+	OnMontageEnded
```
Comment (Ciberusps) [inline Source/UnrealHelperLibrary/Public/AI/Tasks/BTT_PlayAnimMontage.h:26]: 🩷

**Comment by Ciberusps** [comment]
Sry a I'm busy will check on saturday (10% chance - tomorrow 🙃)

**Comment by Ciberusps** [comment]
Testing on `Plugins/UnrealHelperLibrary/Content/Gyms/Gym_PlayAnimMontage_StressTest.umap`
Don't know why but only last character continue to play montages others stops after one montage
image

**Review by Ciberusps — DISMISSED** [review]

File: Source/UnrealHelperLibrary/Private/AI/Tasks/BTT_PlayAnimMontage.cpp [hunk 2]
```diff
@@ -13,92 +12,156 @@ UBTT_PlayAnimMontage::UBTT_PlayAnimMontage(const FObjectInitializer& ObjectIniti
 	: Super(ObjectInitializer)
 {
 	NodeName = "PlayAnimMontage";
-    // TODO try to find the way to remove instancing
-    // instantiating to be able to use Timers
-    bCreateNodeInstance = true;
```
Comment (Ciberusps) [inline Source/UnrealHelperLibrary/Private/AI/Tasks/BTT_PlayAnimMontage.cpp:18]: return instancing and I'll merge?

Comment (nozomanai) [reply Source/UnrealHelperLibrary/Private/AI/Tasks/BTT_PlayAnimMontage.cpp:18]: OK

**Review by Ciberusps — APPROVED** [review]

**Pull request merged**